tools/generate_tables_sql_and_testNames.php

```diff
@@ -23,96 +30,210 @@
+        $fp,
+        "REPLACE INTO test_names (Test_name, Full_name, Sub_group) "
+        ."VALUES ('$tablename', '$title', 1);\n"
+    );
+
+    foreach ($pages as $pageNo => $page) {
+        fwrite(
+            $fp,
+            "INSERT INTO instrument_subtests (Test_name, Subtest_name, Description)"
+            . " VALUES ('$tablename', '" . $tablename  . "_page" . ($pageNo+1)
+            . "', '$page');\n"
+        );
     }
+
     fclose($fp);
 }
 
-echo "\n\Table SQL Generation complete\n";
+// Setup Database object.
+$config    =& \NDB_Config::singleton();
+$db_config = $config->getSetting('database');
+$db        = \Database::singleton(
+    $db_config['database'],
+    $db_config['quatUser'],
+    $db_config['quatPassword'],
+    $db_config['host']
+);
+
+if ($db->isConnected()) {
+    if (!$db->tableExists($sql_query_table_name)) {
+        // Table doesn't exist!
+        $sql_query_statement .= implode($sql_query_columns);
+        $sql_query_statement .=
+            "PRIMARY KEY  (`CommentID`)\n) ENGINE=InnoDB DEFAULT CHARSET=utf8;\n";
+
+        $statement = $db->prepare($sql_query_statement);
+        $statement->execute();
+    }
 
-function enumizeOptions($options, $table, $name){
-    $options=explode("{-}",$options);
-    foreach($options as $option){
-        $option=explode("=>",$option);
-        if($option[0]!='NULL'){
-            $enum[]=$option[0];
+    $statement = $db->prepare(
+        'REPLACE INTO test_names (Test_name, Full_name, Sub_group) VALUES (?, ?, 1)'
+    );
+    $statement->bindParam(1, $tablename, PDO::PARAM_STR);
+    $statement->bindParam(2, $title, PDO::PARAM_STR);
+    $statement->execute();
+
+    foreach ($pages as $pageNo => $page) {
+        $statement    = $db->prepare(
+            'INSERT INTO instrument_subtests (Test_name, Subtest_name, Description)'
+            . ' VALUES (?, ?, ?)'
+        );
+        $subtest_name = $tablename  . "_page" . ($pageNo+1);
+        $statement->bindParam(1, $tablename, PDO::PARAM_STR);
+        $statement->bindParam(2, $subtest_name, PDO::PARAM_STR);
+        $statement->bindParam(3, $page, PDO::PARAM_STR);
+        $statement->execute();
+    }
+}
+/**
+ * The enumizeOptions function
+ *
+ * @param array  $options the enum values.
+ * @param string $table   the name used.
+ * @param string $name    the table uses.
+ *
+ * @return string
+ */
+function enumizeOptions($options, $table, $name)
+{
+    $options = explode("{-}", $options);
+    foreach ($options as $option) {
+        $option = explode("=>", $option);
+        if ($option[0] != 'NULL') {
+            $enum[] = $option[0];
         }
     }
-    if(!is_array($enum)){
+    if (!is_array($enum)) {
         echo "$table $name $options\n";
     }
-    $enum=implode(",",$enum);
+    $enum = implode(",", $enum);
     return "enum($enum)";
 }
-
+/**
+ * The sanitizeSqlString function
+ *
+ * @param string $name to sanitize.
+ *
+ * @return string
+ */
+function sanitizeSqlString($name)
```

Comment:
I'm not sure how this sanitizes a string, exactly. Also, I think it's better not to reinvent the wheel here and instead used prepared statements -- at the very least it is probably better to use built in PHP functions to do this.

Reply:
Unable to bind "table names" or "column names" when using prepared statements. If attempted the column names will have single quotes. I'm using the sanitizeSqlString for when creating the table with the user supplied input for that part.

Reply:
I think I'd prefer in this case to use the custom `escape` function in the DB. It serves the same purpose as this function and follows the standard set by `mysql_real_escape_string` and so will be simpler to maintain going forward. https://github.com/aces/Loris/blob/minor/php/libraries/Database.class.inc#L1315